### PR TITLE
Bug 1210305 Lock screen implementation

### PR DESCRIPTION
**Pull request by harman-ankitgupta**
Implemented the lockscreen as per the latest UX. Please review it

### CODE REVIEW

**Comment by harman-ankitgupta** [comment]
@pgmanutd @Ashish-Developer . Please review it.

**Comment by TatianaMigacheva** [comment]
Please change "commit message" to "Bug 1210305 Lock screen implementation". This is Mozilla template.

**Comment by AlekoNN** [comment]
@harman-red-square/harman-reviewers

**Comment by egorlevichev** [comment]
are changes in 'notifications.js' really required for this bug implementation? From first look they seems more like result of wrong merge/rebase, rather than LS feature implementation

**Comment by harman-ankitgupta** [comment]
As per the requirement, similar kind of notification should be clubbed together and notification node is created in the notifications.js. So for combining the notifications, changes in notifications.js should also be done.
Refer UX id: 2.2 for the same.
UX link : http://r1l8an.axshare.com/#p=2_2_open_app_from_lock_screen

File: apps/system/js/notifications.js [hunk 1]
```diff
@@ -137,7 +137,7 @@ var NotificationScreen = {
             }
             break;
           case 'desktop-notification-close':
-            this.removeNotification(detail.id);
+            this.removeNotification(detail.id, detail.appName);
```
Comment (egorlevichev) [inline apps/system/js/notifications.js]: 1) It's not clear how it's expected to work, if signature of function 'removeNotification()' still have only 1 parameter?

File: apps/system/js/notifications.js [hunk 2]
```diff
@@ -380,37 +380,41 @@ var NotificationScreen = {
   },
 
   addNotification: function ns_addNotification(detail) {
-    // LockScreen window may not opened while this singleton got initialized.
-    this.lockScreenContainer = this.getLockScreenContainer();
+  // LockScreen window may not opened while this singleton got initialized.
```
Comment (egorlevichev) [inline apps/system/js/notifications.js]: This 'change' violates Mozilla coding style requirements. And it seems not required for feature implementation, since only formatting was changed. You shall avoid unnecessary changes in your commits, since it makes review of actual changes not trivial.

**Comment by egorlevichev** [comment]
please fix inline comments, note that comment regarding changed line indents is common for this patch.

**Comment by harman-ankitgupta** [comment]
We have done the method-level line indentation instead of doing it on complete file using two-spaces indent for the same.
If we have to use any other coding style standard,kindly share it with us.

**Comment by egorlevichev** [comment]
Since we will contribute all created code to the Mozilla, we shall follow theirs coding style defined here: https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/Understanding_the_Gaia_codebase#Gaia_coding_style which is mostly follows Google's one: http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml

File: apps/system/js/notifications.js [hunk 3]
```diff
@@ -672,7 +687,8 @@ var NotificationScreen = {
 
   closeNotification: function ns_closeNotification(notificationNode) {
     var notificationId = notificationNode.dataset.notificationId;
-    this.removeNotification(notificationId);
+    var id = notificationNode.id;
+    this.removeNotification(notificationId, id);
```
Comment (davidflanagan) [inline apps/system/js/notifications.js]: It looks like many of the changes in this file are unnecessary changes to the indentation level. This makes it very hard to review. Please fix that before requesting further review.

File: apps/system/lockscreen/js/lockscreen_inputpad.js [hunk 4]
```diff
@@ -147,7 +147,8 @@
   };
 
   LockScreenInputpad.prototype.updatePassCodeUI =
-  function() {
+  function(keyCode) {
```
Comment (davidflanagan) [inline apps/system/lockscreen/js/lockscreen_inputpad.js:150]: I haven't figured out where this keycode is coming from. Where's the code that listens to key events from the keypad?  And are you going to be removing the unnecessary code for displaying and a keypad and handling events on it?

Comment (davidflanagan) [inline apps/system/lockscreen/js/lockscreen_inputpad.js:150]: That doesn't have to be part of this patch, but I need more context information to know what you're doing here so I know what is in scope and what is out of scope for this patch.

File: apps/system/lockscreen/js/lockscreen_inputpad.js [hunk 5]
```diff
@@ -160,19 +161,32 @@
     }
     var i = 4;
     while (i--) {
-      var span = this.passcodeCode.childNodes[i];
-      if (span) {
-        if (this.states.passCodeEntered.length > i) {
-          span.dataset.dot = true;
+    var span = this.passcodeCode.childNodes[i];
```
Comment (davidflanagan) [inline apps/system/lockscreen/js/lockscreen_inputpad.js]: Your indentation is off here, which makes the code very hard to read

File: apps/system/lockscreen/js/lockscreen_inputpad.js [hunk 6]
```diff
@@ -180,10 +194,11 @@
...
         this.dispatchEvent(new window.CustomEvent(
-          'lockscreen-keypad-input', { detail: {
-            key: key
-          }
-        }));
+          'lockscreen-keypad-input', {
+            detail: {
+              key: key
+            }
+          }));
```
Comment (davidflanagan) [inline apps/system/lockscreen/js/lockscreen_inputpad.js]: It appears that there is no change here except to formatting. Your formatting is better, but please don't do that if there are not actually any required changes, because it just makes it harder to review and increases the chances for merge conflicts.

**Comment by AlekoNN** [comment]
Label is removed since new commit has been added.

**Comment by Ashish-Developer** [comment]
label:pr_review_passed

**Comment by AlekoNN** [comment]
Let's make the commit towards gaia 2.2r branch and solve the comments there if necessary.
